src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -97,18 +97,172 @@ async function pickProject(
+    await vscode.commands.executeCommand('markdown.showPreview', vscode.Uri.file(planFilePath))
+    transformByQState.setPlanFilePath(planFilePath)
+    await vscode.commands.executeCommand('setContext', 'gumby.isPlanAvailable', true)
+    throwIfCancelled()
+}
+
+export async function pollTransformationTillComplete(jobId: string) {
+    let status = ''
+    try {
+        status = await pollTransformationJob(jobId, CodeWhispererConstants.validStatesForCheckingDownloadUrl)
+    } catch (error) {
+        errorMessage = 'Failed to get transformation job status'
+        getLogger().error(errorMessage, error)
+        throw new ToolkitError(errorMessage, { cause: error as Error })
+    }
+
+    return status
+}
+
+export async function setTransformationJobStatus(status: string) {
+    if (!(status === 'COMPLETED' || status === 'PARTIALLY_COMPLETED')) {
+        errorMessage = 'Failed to complete transformation'
+        getLogger().error(errorMessage)
+        sessionPlanProgress['transformCode'] = StepProgress.Failed
+
+        throw new ToolkitError(errorMessage, { code: 'JobDidNotSucceed' })
+    }
+
+    sessionPlanProgress['transformCode'] = StepProgress.Succeeded
+    transformByQState.setToSucceeded()
+    if (status === 'PARTIALLY_COMPLETED') {
+        transformByQState.setToPartiallySucceeded()
+        resultStatusMessage = 'JobPartiallySucceeded'
+    } else {
+        resultStatusMessage = 'JobCompletedSuccessfully'
+    }
+
+    await vscode.commands.executeCommand('aws.amazonq.transformationHub.reviewChanges.reveal')
+    await vscode.commands.executeCommand('aws.amazonq.refresh')
+    sessionPlanProgress['returnCode'] = StepProgress.Succeeded
+}
+
 export async function startTransformByQ() {
+    // 1: Validate inputs. If failed, Error will be thrown and execution stops
+    const userInputState = await validateTransformJob()
+
+    // Set the default state variables for our store and the UI
+    await setModernizationToRunningState(userInputState)
+
+    try {
+        // Set web view UI to poll for progress
+        intervalId = setInterval(() => {
+            void vscode.commands.executeCommand('aws.amazonq.showPlanProgressInHub', startTime.getTime())
+        }, CodeWhispererConstants.progressIntervalMs)
+
+        // step 1: CreateCodeUploadUrl and upload code
+        const uploadId = await preTransformUploadCode(userInputState)
+
+        // step 2: StartJob and store the returned jobId in TransformByQState
+        const jobId = await transformStartJob(uploadId)
+
+        // step 3 (intermediate step): show transformation-plan.md file
+        // TO-DO: on IDE restart, resume here if a job was ongoing
+        await pollTransformationStatusTillPlanReady(jobId)
+
+        // step 4: poll until artifacts are ready to download
+        const status = await pollTransformationTillComplete(jobId)
+
+        await setTransformationJobStatus(status)
+    } catch (error) {
+        await modernizationJobErrorHandler()
```

Comment:
Maybe pass the error into the ErrorHandler and Log the error? I think we have discussed that log is local in the IDE and not sent to the cloud, so should not safe to include original stack trace. If this assumption is not correct please let me know!

Reply:
Absolutely agree. If we log to users local file system this would be okay.